fairscale/experimental/optim/dynamic_loss_scaler.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+To prevent underflow or overflow of gradients, DynamicLossScaler is used to
+dynamically scale up and down gradients by scaling the loss.
```

Comment:
Will be good to mention what is different about this from the [torch.cuda.amp.GradScaler](https://pytorch.org/docs/stable/_modules/torch/cuda/amp/grad_scaler.html#GradScaler) class - i.e it needs to be used for updates on the CPU device